lib/authentication-context.js

```diff
@@ -248,7 +248,8 @@ AuthenticationContext.prototype.acquireToken = function(resource, userId, client
  * @param {string}   clientId                            The OAuth client id of the calling application.
  * @param {AcquireTokenCallback}   callback              The callback function.
  */
-AuthenticationContext.prototype.acquireTokenWithUsernamePassword = function(resource, username, password, clientId, callback)  {
+AuthenticationContext.prototype.acquireTokenWithUsernamePassword = function(resource, username, password, clientId, clientSecret, callback)  {
```

Comment:
@phedinkus I would request you to go through the samples provided in https://github.com/AzureAD/azure-activedirectory-library-for-nodejs/tree/master/sample and see if that solves the issue you are hitting.